### PR TITLE
WIP: SelectedLicense component test

### PR DESCRIPTION
**Pull request by onyijne**
Fixes #65 

**Description**

**Other information**

**Checklist:**
- [ ] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [ ] My pull request targets the `master` branch of the repository.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

### CODE REVIEW

**Comment by onyijne** [comment]
@akmadian kindly see if there are any specific updates you would be looking out for before I remove the WIP. Thanks

File: test/unit/specs/components/SelectedLicense.spec.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { shallowMount } from '@vue/test-utils'
```
Comment (akmadian) [inline test/unit/specs/components/SelectedLicense.spec.js]: - Please test that the link to the full license text is valid. 
 - I would also like to see tests for licenses other than just BY

Comment (akmadian) [reply test/unit/specs/components/SelectedLicense.spec.js]: Please add test for the computed `iconsList` property to be sure that the correct array is being returned based on the shortlicensename. Remember to test this for multiple licenses.

File: test/unit/specs/components/SelectedLicense.spec.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+describe('SelectedLicense.spec.js', () => {
+    let cmp
+
+    beforeEach(() => {
+        cmp = shallowMount(SelectedLicense, {
+        // Be aware that props is overridden using `propsData`
+            propsData: {
+                fullLicenseName: 'Attribution',
+                shortLicenseName: 'BY'
```
Comment (akmadian) [inline test/unit/specs/components/SelectedLicense.spec.js]: Please format shortLicenseName as "CC {name} 4.0". For example, BY would be "CC BY 4.0", BY-ND would be "CC BY-ND 4.0". This is how the prop is actually formatted when passed to the component.

File: test/unit/specs/components/SelectedLicense.spec.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import SelectedLicense from '@/components/SelectedLicense.vue'
+
+describe('SelectedLicense.spec.js', () => {
+    let cmp
+
+    beforeEach(() => {
+        cmp = shallowMount(SelectedLicense, {
+        // Be aware that props is overridden using `propsData`
+            propsData: {
+                fullLicenseName: 'Attribution',
```
Comment (akmadian) [inline test/unit/specs/components/SelectedLicense.spec.js]: Please format fullLicenseName as "{name} 4.0 International". For example, BY would be "Attribution 4.0 International", BY-ND would be "Attribution-NoDerivatives 4.0 International". This is how the prop is actually formatted when passed to the component.

File: test/unit/specs/components/SelectedLicense.spec.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+                shortLicenseName: 'BY'
+            }
+        })
+    })
+
+    it('has received \'Attribution\' as the fullLicenseName property', () => {
+        expect(cmp.vm.fullLicenseName).toEqual('Attribution')
```
Comment (akmadian) [inline test/unit/specs/components/SelectedLicense.spec.js]: Please update in accordance with correct prop format

File: test/unit/specs/components/SelectedLicense.spec.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    })
+
+    it('has received \'BY\' as the shortLicenseName property', () => {
+        expect(cmp.vm.shortLicenseName).toEqual('BY')
```
Comment (akmadian) [inline test/unit/specs/components/SelectedLicense.spec.js]: Please update in accordance with correct prop format

File: test/unit/specs/components/SelectedLicense.spec.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    })
+
+    it('has the expected html structure', () => {
+        expect(cmp.element).toMatchSnapshot()
```
Comment (akmadian) [inline test/unit/specs/components/SelectedLicense.spec.js]: This test is ok for BY, but not necessary for other licenses.

File: test/unit/specs/components/SelectedLicense.spec.js (hunk 1 above)
Comment (akmadian) [reply test/unit/specs/components/SelectedLicense.spec.js]: Please add test for `watch` to be sure that the link to the full license text updates properly when the shortlicensename property updates.

Comment (akmadian) [reply test/unit/specs/components/SelectedLicense.spec.js]: Bonus points if you can find a way to check that the URL isn't dead :)

**Comment by akmadian** [comment]
Closing because this PR is now obsolete because of changes to src

**Pull request closed without merging**